src/components/Pagination/Pagination.tsx

```diff
@@ -2,0 +1,78 @@
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+type PropsPagination = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination = ({
+  total,
+  perPage,
+  currentPage = 1,
+  onPageChange,
+}: PropsPagination) => {
+  const totalPages = Math.ceil(getNumbers(1, total).length / perPage);
+  const totalPagesArr = getNumbers(1, totalPages);
+
+  return (
+    <ul className="pagination">
+      <li
+        className={cn(
+          'page-item',
+          { disabled: currentPage === 1 },
+          { active: currentPage === 1 },
+        )}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={() => {
+            if (currentPage !== 1) {
+              onPageChange(currentPage - 1);
+            }
+          }}
+        >
+          «
+        </a>
+      </li>
+      {totalPagesArr.map(p => (
+        <li className={cn('page-item', { active: currentPage === p })} key={p}>
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${p}`}
```

Comment:
Using `href="#${p}"` will change the URL fragment when clicked. This could lead to unwanted behavior in a single-page application. It's better to prevent the default action of the click event to avoid changing the URL fragment.